### ISSUE TITLE
Update failure for Ubuntu package

### ISSUE THREAD

**Issue by monetdb-team** [MonetDB/MonetDB#3001]
Date: 2010-11-15 22:41:13 +0100
From: Lieuwe &lt;<lieuwer>&gt;
To:   SQL devs &lt;<bugs-sql>&gt;
Version: 2.38.5 (Jun2010-SP2) [obsolete]
CC:   sjoerd

Last updated: 2010-12-20 09:14:34 +0100

## Comment 15199
Date: 2010-11-15 22:41:13 +0100
From: Lieuwe &lt;<lieuwer>&gt;

User-Agent:       Mozilla/5.0 (X11; U; Linux i686; en-US; rv:1.9.2.12) Gecko/20101027 Ubuntu/10.04 (lucid) Firefox/3.6.12
 Build Identifier: 
 
 After an attempted and failed update of the monetdb5-sql package, the sql front-end is no longer working properly. The error returned states: "E: /var/cache/apt/archives/monetdb5-sql_2.40.1-20101110_i386.deb: subprocess new pre-removal script returned error exit status 1"
 I am also no longer able to uninstall or re-install the package. I can still run merovingian, but when running some simple python code that connect to a db I get the following error:
 Traceback (most recent call last):
   File "tryMSEED.py", line 18, in <module>
     hostname='localhost', database='seedtest')
   File "/usr/lib/python2.6/monetdb/sql/__init__.py", line 27, in connect
     return Connection(*args, **kwargs)
   File "/usr/lib/python2.6/monetdb/sql/connections.py", line 52, in __init__
     password=password, database=database, language="sql")
   File "/usr/lib/python2.6/monetdb/mapi2.py", line 80, in connect
     self.__login()
   File "/usr/lib/python2.6/monetdb/mapi2.py", line 87, in __login
     challenge = self.__getblock()
   File "/usr/lib/python2.6/monetdb/mapi2.py", line 226, in __getblock
     len(flag))
 monetdb.monetdb_exceptions.OperationalError: server returned 0 bytes, I need 2
 
 
 Reproducible: Always
 
 ### Steps to Reproduce:
 1. I didn't pay enough attention during the first update, but now it happens every time
 2.
 3.
 ### Actual Results:  
 Failure to remove package (for update or otherwise) and current package is not functioning
 
 ### Expected Results:  
 Update the package

## Comment 15200
Date: 2010-11-16 12:55:50 +0100
From: Sjoerd Mullender &lt;<sjoerd>&gt;

This is indeed a bug, but you can work around it:
 First edit the file /etc/default/monetdb5-sql and change the value for STARTUP to "yes".
 Then start the monetdb5-sql service by running
     sudo /etc/init.d/monetdb5-sql start
 After this, redo the upgrade.  It should now correctly stop the service (which is where it went wrong the first time), upgrade the packages, and restart the service.
 Finally, you can undo the first two changes by editing the file again and by running
     sudo /etc/init.d/monetdb5-sql stop

## Comment 15201
Date: 2010-11-16 13:01:26 +0100
From: Sjoerd Mullender &lt;<sjoerd>&gt;

Changeset [6585656d5f30](https://dev.monetdb.org/hg/MonetDB?cmd=changeset;node=6585656d5f30)  made by Sjoerd Mullender <sjoerd@acm.org> in the MonetDB repo, refers to this bug.
 
 For complete details, see [http//devmonetdborg/hg/MonetDB?cmd=changeset;node=6585656d5f30](https://dev.monetdb.org/hg/MonetDB?cmd=changeset;node=http//devmonetdborg/hg/MonetDB?cmd=changeset;node=6585656d5f30) 
 
 Changeset description:
 
 	Don't fail when trying to stop an already stopped daemon.
 	This should fix bug #2723.

## Comment 15202
Date: 2010-11-16 13:03:05 +0100
From: Sjoerd Mullender &lt;<sjoerd>&gt;

The real fix is to not fail an upgrade when the monetdb5-sql daemon isn't running.  This is now implemented and will be in the next release.

## Comment 15203
Date: 2010-11-16 17:12:51 +0100
From: Lieuwe &lt;<lieuwer>&gt;

I'm not certain whether this is a new bug, or simply a result from my initial failed update, but somehow the workaround is not working out for me.
 I had already changed the setting before and after trying to run the given command I get the following feedback:
 Starting MonetDB SQL server: unable to open .merovingian_pass: Permission denied
 lieuwe@lieuwe-laptop:~$ fatal startup condition encountered, aborting startup
 Merovingian 1.3 stopped
  
 I figured running merovingian itself might help and after doing so I get the feedback:
 Starting MonetDB SQL server: /usr/bin/merovingian already running.
 
 Strangely enough, when I now try to stop monetdb-sql I get the following feedback:
 Stopping MonetDB SQL server: No /usr/bin/merovingian found running; none killed.
 
 I guess that this is the reason why my update won't work. Somehow it is running, but somehow it isn't at the same time?

## Comment 15204
Date: 2010-11-16 17:34:10 +0100
From: Sjoerd Mullender &lt;<sjoerd>&gt;

Sounds like a permissions problem.
 
 Did you run merovingian under your own ID before?  By starting the server using the script in /etc/init.d, merovingian is started with the ID of user monetdb, group monetdb.  If you had run it before with your own ID, the files /var/MonetDB5/dbfarm/.merovingian* are probably owned by you, so the server doesn't have sufficient permissions.
 
 The easiest solution is probably to run merovingian as a system daemon (i.e. keep the STARTUP=yes in /etc/default/monetdb5-sql) and change ownership of everything under /var/MonetDB5: sudo chown -R monetdb:monetdb /var/MonetDB5 .
 
 Alternatively:
 
 If the goal is just to finish the upgrade, you can temporarily move the /var/MonetDB5 directory to the side (after stopping merovingian) and create one using
 sudo mkdir /var/MonetDB5
 sudo chown monetdb:monetdb /var/MonetDB5
 sudo chmod 770 /var/MonetDB5
 
 I'm not sure whether you also need a dbfarm directory underneath.
 
 Then do the upgrade per my earlier instructions.
 
 Afterwards, you can stop merovingian, remove /var/MonetDB5 and move the old directory back.

## Comment 15205
Date: 2010-11-17 10:09:21 +0100
From: Sjoerd Mullender &lt;<sjoerd>&gt;

An easier workaround is this:
 Edit the file /etc/init.d/monetdb5-sql, and change the line
 start-stop-daemon --stop --pidfile $PIDFILE --exec $DAEMON
 to read
 running && start-stop-daemon --stop --pidfile $PIDFILE --exec $DAEMON
 In other words, add "running &&" to the front of that line.
 
 This seems to work also when you had started merovingian under your own ID before since there will then no need to start merovingian at all under the monetdb ID.  This is what gave the permissions problems.
